### PR TITLE
Performance page

### PR DESCRIPTION
**Pull request by insectengine**
1) Add new performance page
2) Add to index page (re-org features band)
3) Add performance page to primary navigation (re-org navigation)
4) Add link to continuum page (Reactive) to the primary navigation

@holly-cummins

### CODE REVIEW

**Comment by holly-cummins** [comment]
+1 to having this page, But I've made some suggestions on Jeff Beck's google document that it would be nice to see incorporated here. The version on this PR doesn't mention our excellent throughput, which we need to start talking about.

**Comment by insectengine** [comment]
@holly-cummins - please check this out again as I've updated the content.

**Comment by holly-cummins** [comment]
Looks good to me, thanks! :) 

I'm looking forward to using that throughput icon in other places, too.

**Comment by holly-cummins** [comment]
I'd add an approving review but I don't have the permissions to assign reviewers.

**Comment by insectengine** [comment]
I added some reviews.

File: _includes/header-navigation.html [hunk 1]
```diff
@@ -15,10 +15,11 @@
         <ul class="submenu">
           <li><a href="{{site.baseurl}}/about" class="{% if page.url contains '/about/' %}active{% endif %}">WHAT IS QUARKUS?</a></li>
           <li><a href="{{site.baseurl}}/container-first" class="{% if page.url contains '/container-first/' %}active{% endif %}">CONTAINER FIRST</a></li>
-          <li><a href="{{site.baseurl}}/continuum" class="{% if page.url contains '/continuum/' %}active{% endif %}">VERSATILITY</a></li>
+          <li><a href="{{site.baseurl}}/performance" class="{% if page.url contains '/performance/' %}active{% endif %}">PERFORMANCE</a></li>
           <li><a href="{{site.baseurl}}/developer-joy" class="{% if page.url contains '/developer-joy/' %}active{% endif %}">DEVELOPER JOY</a></li>
           <li><a href="{{site.baseurl}}/kubernetes-native" class="{% if page.url contains '/kubernetes-native/' %}active{% endif %}">KUBERNETES NATIVE</a></li>
           <li><a href="{{site.baseurl}}/standards" class="{% if page.url contains '/standards/' %}active{% endif %}">STANDARDS</a></li>
+          <li><a href="{{site.baseurl}}/continuum" class="{% if page.url contains '/continuum/' %}active{% endif %}">REACTIVE</a></li>
```
Comment (cescoffier) [inline _includes/header-navigation.html:22]: No, we definitely don't want to have REACTIVE here, the page is NOT about reactive.

File: _includes/header-navigation.html [hunk 2]
```diff
@@ -15,10 +15,11 @@
         <ul class="submenu">
           <li><a href="{{site.baseurl}}/about" class="{% if page.url contains '/about/' %}active{% endif %}">WHAT IS QUARKUS?</a></li>
           <li><a href="{{site.baseurl}}/container-first" class="{% if page.url contains '/container-first/' %}active{% endif %}">CONTAINER FIRST</a></li>
-          <li><a href="{{site.baseurl}}/continuum" class="{% if page.url contains '/continuum/' %}active{% endif %}">VERSATILITY</a></li>
+          <li><a href="{{site.baseurl}}/performance" class="{% if page.url contains '/performance/' %}active{% endif %}">PERFORMANCE</a></li>
```
Comment (cescoffier) [inline _includes/header-navigation.html:18]: Why did you move the continuum page to the end, and put performance here?

File: _includes/homepage-features-icon-band.html [hunk 3]
```diff
@@ -12,16 +12,16 @@ <h3><a href="/container-first">Container First</a></h3>
       <p>Quarkus tailors your application for GraalVM and HotSpot. Amazingly fast boot time, incredibly low RSS memory (not just heap size!) offering near instant scale up and high density memory utilization in container orchestration platforms like Kubernetes. We use a technique we call compile time boot.</p>
     </div>
     <div class="width-4-12 width-12-12-m contrib-block">
-      <img class="light-only" src="{{site.baseurl}}/assets/images/about/icon-reactive.svg">
-      <img class="dark-only" src="{{site.baseurl}}/assets/images/about/icon-reactive-dark.svg">
-      <h3><a href="/continuum">Reactive Core</a></h3>
-      <p>Built on a robust reactive core, Quarkus ensures fast and efficient performance, supporting the development of a wide variety of modern applications.</p>
+      <img class="light-only" src="{{site.baseurl}}/assets/images/performance/icon-performance.svg">
```
Comment (cescoffier) [inline _includes/homepage-features-icon-band.html:15]: I would not have put performance in the second item, on the second line. 

I know it's a core strength, but, that's not going to attract lots of developers (otherwise, many languages would not be as popular as they are...).

File: _includes/homepage-features-icon-band.html [hunk 4]
```diff
@@ -12,16 +12,16 @@ <h3><a href="/container-first">Container First</a></h3>
...
+      <img class="dark-only" src="{{site.baseurl}}/assets/images/performance/icon-performance-dark.svg">
+      <h3><a href="/performance">Performance</a></h3>
+      <p>Quarkus streamlines framework optimizations in the build phase to reduce runtime dependencies and improve efficiency. By precomputing metadata and optimizing class loading, it ensures fast startup times for JVM and native binary deployments, cutting down on memory usage.</p>
```
Comment (cescoffier) [inline _includes/homepage-features-icon-band.html:18]: Hum, this is semantically equivalent to the last sentence of the container feature.

File: _includes/performance.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
+<div class="full-width-bg component">
+  <div class="grid-wrapper">
+    <div class="width-3-12 width-12-12-m img-vert-center">
+      <img class="light-only" src="{{site.baseurl}}/assets/images/performance/icon-performance.svg" alt="Performance icon">
+      <img class="dark-only" src="{{site.baseurl}}/assets/images/performance/icon-performance-dark.svg" alt="Performance icon">
+    </div>
+    <div class="width-9-12 width-12-12-m">
+      <h2>The Quarkus Way</h2>
```
Comment (cescoffier) [inline _includes/performance.html:8]: HTML makes reviewing very hard. Could you try the technique I used for continuum?

File: _includes/performance.html [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      <p>All that code is run only during the compilation phase; the classes don't even get loaded when  the app is started. This means the JIT can make much better optimisations, and spend less memory in doing so.</p>
+      <p>By performing these optimizations at build time, Quarkus ensures that the heavy lifting is done once, not at every startup, reducing memory usage and startup time.</p>
+      <p>This approach benefits both GraalVM native images and traditional HotSpot JVM deployments, leading to faster, smaller, and more resource-efficient Java applications.</p>
+    </div>
+    <div class="width-3-12 width-12-12-m img-vert-center">
+      <img class="light-only" src="{{site.baseurl}}/assets/images/performance/icon-memory.svg" alt="Memory icon">
+      <img class="dark-only" src="{{site.baseurl}}/assets/images/performance/icon-memory-dark.svg" alt="Memory icon">
+    </div>
+    <div class="width-9-12 width-12-12-m">
+      <h2>Reduced Memory</h2>
```
Comment (cescoffier) [inline _includes/performance.html:26]: Should be h3 - it's part of the quarkus way, which actually is named" build-time principle.

File: _includes/performance.html [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+      <p>Quarkus reduces memory utilization for both traditional JVM deployments as well as native binaries using Ahead-of-Time (AOT) Compilation with GraalVM. In a traditional architecture many classes are loaded into memory, only to be pruned out later. Quarkus is able to avoid this wasted classloading. Quarkus also leverages build-time metadata processing, lowering memory consumption during execution.</p>
+    </div>
+    <div class="width-3-12 width-12-12-m img-vert-center">
+      <img class="light-only" src="{{site.baseurl}}/assets/images/performance/icon-startup.svg" alt="Startup icon">
+      <img class="dark-only" src="{{site.baseurl}}/assets/images/performance/icon-startup-dark.svg" alt="Startup icon">
+    </div>
+    <div class="width-9-12 width-12-12-m">
+      <h2>Fast Startup Time</h2>
```
Comment (cescoffier) [inline _includes/performance.html:34]: Same, h3

File: _includes/performance.html [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    </div>
+    <div class="width-3-12 width-12-12-m img-vert-center">
+      <img class="light-only" src="{{site.baseurl}}/assets/images/performance/icon-throughput.svg" alt="Throughput icon">
+      <img class="dark-only" src="{{site.baseurl}}/assets/images/performance/icon-throughput-dark.svg" alt="Throughput icon">
+    </div>
+    <div class="width-9-12 width-12-12-m">
+      <h2>Higher Throughput</h2>
+      <p>Doing more work upfront at build time doesn’t just improve memory footprint and startup times. It also makes normal program execution faster. How? The output of a Quarkus build process is JIT-friendly, which means the JIT can make better optimizations, and get the application to a super-optimized state faster. For example, because there’s less unused bytecode in a Quarkus application, the JIT can inline more effectively. Early elimination of unused classes also enables the JVM to use monomorphic method dispatching, instead of the much slower megamorphic method dispatching. Megamorphic dispatching is necessary when there are several implementations of the same interface present on the classpath.</p>
```
Comment (cescoffier) [inline _includes/performance.html:44]: This is not the reason, it's only a sub-part of the reason. The main reason is the reactive core. This is great, but as the cherry on the top, the cake is the reactive core.

File: _includes/performance.html [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    </div>
+    <div class="width-3-12 width-12-12-m img-vert-center">
+      <img class="light-only" src="{{site.baseurl}}/assets/images/performance/icon-diskspace.svg" alt="Disk footprint icon">
+      <img class="dark-only" src="{{site.baseurl}}/assets/images/performance/icon-diskspace-dark.svg" alt="Disk footprint icon">
+    </div>
+    <div class="width-9-12 width-12-12-m">
+      <h2>Disk Footprint</h2>
+      <p>Quarkus reduces the footprint of Java applications by employing build-time processing to eliminate unnecessary runtime dependencies and by optimizing the application’s deployment artifacts. It packages only the essential classes and resources needed at runtime, removing unused code through techniques like dead code elimination. When using GraalVM for native compilation, Quarkus further reduces the footprint by compiling the application into a compact native binary, stripping out the JVM and related dependencies.</p>
```
Comment (cescoffier) [inline _includes/performance.html:52]: I would hide this.

Our disk footprint is 10 times bigger than the equivalent Go application.

Comment (cescoffier) [reply _includes/performance.html:52]: (and compressing native executable comes with a cost)

**Review by cescoffier — CHANGES_REQUESTED** [review]

**Comment by insectengine** [comment]
I feel this is ready for review...
**1) Home Page** - Updated the "features" band to "benefits" and added the Performance icon/text
https://quarkus-site-pr-2122-preview.surge.sh/

**2) Performance Page** - Updated content from the doc.
https://quarkus-site-pr-2122-preview.surge.sh/performance/

**3) Flyout Nav** - Added to the "Why" flyout

**Comment by holly-cummins** [comment]
Where are we with this, @cescoffier? It would be fantastic to get this reviewed and up so we can start getting eyeballs on this important content.

**Comment by cescoffier** [comment]
@holly-cummins @insectengine Hey! Sorry, missed the last update. It looks A LOT better. I will do a few suggestion later today:

- @holly-cummins, don't we have an image for the introduction? I preferred ending the paragraph with the `Why`
- The is the weird (or absence of) transition in the reactive core section
- The high throughput paragraph does not talk about high throughput per se.

File: _includes/performance.html [hunk 10]
```diff
@@ -0,0 +1,91 @@
+<div class="full-width-bg component">
+  <div class="grid-wrapper">
+      <div class="width-3-12 width-12-12-m">
+        <img class="light-only" src="{{site.baseurl}}/assets/images/performance/icon-performance.svg" alt="Container image">
+        <img class="dark-only" src="{{site.baseurl}}/assets/images/performance/icon-performance-dark.svg" alt="Container image">
+      </div>
+      <div class="width-9-12 width-12-12-m">
+        <p class="intropara">Quarkus is engineered for efficiency. It  leverages build-time optimizations and a reactive core to achieve high throughput, fast startup times, low response latency, and minimal resource consumption. In other words, Quarkus is fast, and here is why; <strong>Reduced Memory Footprint, Fast startup Time and Higher Throughput</strong></p>
```
[the remark this answers is not in the document]
Comment (holly-cummins) [reply _includes/performance.html]: +1, I like James revision. 

We had a few pictures in the draft at various stages, but I think maybe the one you mean is a mockup I made with graph lines. I think if we go that way we ought to get some 'real' data from @johnaohara and then abstract that into a graphic. It wouldn't be 'accurate' in the same way an actual graph would be, but it would have some foundation in a point-in-time reality. 

But I suggest we get the content up and then iterate on the visuals.

**Comment by holly-cummins** [comment]
@cescoffier, it looks like your suggested changes are now live at https://quarkus-site-pr-2122-preview.surge.sh/performance/

**Comment by cescoffier** [comment]
@holly-cummins not the landing page (order + outdated content).

Maybe we can limit this PR to the performance page.

**Comment by cescoffier** [comment]
@insectengine Thanks for the update, waiting for the previous (because, no... I'm not reviewing HTML blurb :-))

**Comment by cescoffier** [comment]
@insectengine The content on the landing page is outdated:

That's what we have on quarkus.io:
image

**Comment by insectengine** [comment]
@cescoffier - updated the Reactive title/content for the homepage.

**Review by cescoffier — APPROVED** [review]

**Comment by cescoffier** [comment]
LGTM, @holly-cummins Are you ok? 

@insectengine maybe squash the commits. 12 commits, that's a bit too much :-)

**Review by holly-cummins — APPROVED** [review]

**Comment by insectengine** [comment]
A merge commit blocked any squashing I tried to do. I pulled the changed files into a new branch and submitted a new PR for the Performance page.

https://github.com/quarkusio/quarkusio.github.io/pull/2163
